driver/compiler_hooks.ml

```diff
@@ -121,6 +124,10 @@ let register : type a. a pass -> (a -> unit) -> unit =
   | Cfg -> hooks.cfg <- f :: hooks.cfg
   | Cmm -> hooks.cmm <- f :: hooks.cmm
   | Inlining_tree -> hooks.inlining_tree <- f :: hooks.inlining_tree
+  | Check_allocations ->
+    Flambda_backend_flags.alloc_check := true;
```

Comment:
I think you're right, I've removed this line. Should we replace it by `assert Flambda_backend_flags.alloc_check` to ensure that the user has provided the flag when using this hook? Or is it fine to silently fail in that case?

Reply:
I am not sure exactly how the hooks will
be used, but in general failing silently
might not be user friendly. If the "wrapper"
using the hooks is responsible for passing
the flag, it doesn't seem unreasonable to
fail loudly here (since it is an internal error).
I would use a `Misc.fatal` rather than a mere
`assert`, though.

Reply:
This will now fail when `alloc_check` is not on